src/nbl/video/CVulkanConnection.h

```diff
@@ -42,79 +59,86 @@ class CVulkanConnection final : public IAPIConnection
 
         VkInstanceCreateInfo instanceCreateInfo = { VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO };
         instanceCreateInfo.pNext = (VkDebugUtilsMessengerCreateInfoEXT*)&debugUtilsMessengerCreateInfo;
+        instanceCreateInfo.flags = static_cast<VkInstanceCreateFlags>(0);
         instanceCreateInfo.pApplicationInfo = &applicationInfo;
         instanceCreateInfo.enabledLayerCount = static_cast<uint32_t>(requiredInstanceLayerNames.size());
         instanceCreateInfo.ppEnabledLayerNames = requiredInstanceLayerNames.data();
+        instanceCreateInfo.enabledExtensionCount = 3u;
+        instanceCreateInfo.ppEnabledExtensionNames = instanceExtensions;
 
+        VkInstance vk_instance;
+        if (vkCreateInstance(&instanceCreateInfo, nullptr, &vk_instance) != VK_SUCCESS)
         {
-            // Todo(achal): Get this from the user
-            const bool isWorkloadHeadlessCompute = false;
+            printf("Failed to create vulkan instance, for some reason!\n");
```

Comment:
use the engine logger (constructor should take a logger just like COpenGL_Connection)